service/policy/db/query.sql

```diff
@@ -81,6 +81,154 @@ WHERE (NULLIF(@kas_id, '') IS NULL OR kas.id = @kas_id::uuid)
 GROUP BY 
     kas.id;
 
+-- name: GetAttributeByDefOrValueFqn :one
+-- get the attribute definition for the provided value or definition fqn
+WITH target_definition AS (
+    SELECT ad.id
+    FROM attribute_definitions ad
+    INNER JOIN attribute_fqns af ON af.attribute_id = ad.id
+    WHERE af.fqn = LOWER($1)
```

Comment:
Would it be to much to make this plural by making this an `IN`? This way you could send a single SQL query. Or would this break other bits of the query?

Reply:
A batched query is definitely a good enhancement idea I've also thought about. I'd like to get this merged first to match the existing behavior and detect any potential regressions. I don't think it would be quite as simple as just updating that one query change, and the existing tests not introduced in this PR would also need updates, which I'd like to avoid for the moment. I'll open an issue for this. 

Not that it's not important, but so far our performance constraints have not been policy related (though that could change as we continue to optimize).

Reply:
https://github.com/opentdf/platform/issues/1357